src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -832,8 +832,7 @@ private static bool IsW3CId(string id)
             //  = 55 chars (see https://w3c.github.io/trace-context)
             // The version (00-fe) is used to indicate that this is a WC3 ID.
             return id.Length == 55 &&
-                   ('0' <= id[0] && id[0] <= '9' || 'a' <= id[0] && id[0] <= 'f') &&
-                   ('0' <= id[1] && id[1] <= '9' || 'a' <= id[1] && id[1] <= 'e');
+                   ((HexConverter.FromLowerChar(id[0]) << 4) + HexConverter.FromLowerChar(id[1]) < 255);
```

Comment:
nit: maybe it is a bit more readable if both are bitwise or both are arithmetic operators. with both bitwise, codegen is slightly efficient (used hex 0xfe to match the comment on line above):

```suggestion
                   ((HexConverter.FromLowerChar(id[0]) << 4) | HexConverter.FromLowerChar(id[1])) <= 0xfe;
```